### PR TITLE
Ensure primary-backup client/server threads are cleaned up

### PR DESCRIPTION
**Pull request by kuujo**

### CODE REVIEW

File: protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java [hunk 1]
```diff
@@ -631,6 +631,6 @@ private void changeRole(PrimaryTerm term) {
         future.complete(null);
       }
     });
-    return future;
+    return future.thenRunAsync(() -> threadContext.close());
```
Comment (johnou) [inline protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java]: isn't this closed in PrimaryBackupServerContext..?

Comment (johnou) [reply protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java]: oh context, not the factory.

Comment (kuujo) [reply protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java]: yeah it's redundant

Comment (kuujo) [reply protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java]: Actually, no it's not. If a partition shuts down a server without shutting down the `ThreadContextFactory` then the thread context should still be shutdown here.

**Pull request merged**